controllers/articles.controller.js

```diff
@@ -44,4 +45,17 @@ exports.postArticleComment = (req, res, next) => {
       res.status(201).send({ comment });
     })
     .catch((err) => next(err));
+};
+
+exports.patchArticleByID = (req, res, next) => {
+  const { article_id } = req.params;
+  const { body } = req;
+  return Promise.all([
+    updateArticle(article_id, body),
+    checkArticleExists(article_id),
```

Comment:
You _could_ use a checkExists function here, but what happens if we pass an invalid or not found article_id to the patch request query? Does psql help us out at all?
Consider the benefits from both sides, if we can rely on just the one query to determine happy and sad path logic then we are avoiding two database requests every single time 👍